es-app/src/CollectionSystemManager.cpp

```diff
@@ -18,29 +18,55 @@
 
 namespace fs = boost::filesystem;
 std::string myCollectionsName = "collections";
+std::string defaultEditingCollection = "favorite";
 
 /* Handling the getting, initialization, deinitialization, saving and deletion of
  * a CollectionSystemManager Instance */
 CollectionSystemManager* CollectionSystemManager::sInstance = NULL;
 
-CollectionSystemManager::CollectionSystemManager(Window* window) : mWindow(window)
+CollectionSystemManager::CollectionSystemManager(Window* window) : mWindow(window), mIsEditing(false), mCustomCollectionsBundle(NULL)
+{
+	createCollectionSystemDeclarations();
+
+	getStandardEnvironment();
+
+	std::string path = getCollectionsFolder();
+	if (!fs::exists(path))
+	{
+		fs::create_directory(path);
+	}
+
+	if (mCollectionSystems.find(defaultEditingCollection) != mCollectionSystems.end() &&
+		mCollectionSystems.at(defaultEditingCollection)->isEnabled)
+	{
+		setEditMode(defaultEditingCollection);
+	}
+}
+
+void CollectionSystemManager::createCollectionSystemDeclarations()
 {
 	CollectionSystemDecl systemDecls[] = {
-		//type                  name            long name            //default sort              // theme folder            // isCustom
-		{ AUTO_ALL_GAMES,       "all",          "all games",         "filename, ascending",      "auto-allgames",           false },
-		{ AUTO_LAST_PLAYED,     "recent",       "last played",       "last played, descending",  "auto-lastplayed",         false },
-		{ AUTO_FAVORITES,       "favorites",    "favorites",         "filename, ascending",      "auto-favorites",          false },
-		{ CUSTOM_COLLECTION,    myCollectionsName,  "collections",    "filename, ascending",      "custom-collections",      true }
+		//type               name               long name            default sort                theme folder               isCustom     isEditable
+		{ AUTO_ALL_GAMES,    "all",             "all games",         "filename, ascending",      "auto-allgames",           false,       false },
+		{ AUTO_LAST_PLAYED,  "recent",          "last played",       "last played, descending",  "auto-lastplayed",         false,       false },
+		{ AUTO_FAVORITES,    "favorite",        "favorites",         "filename, ascending",      "auto-favorites",          false,       true },
+		//{ AUTO_HIDDEN,       "hidden",          "hidden",            "filename, ascending",      "custom-collections",      false,       true },
```

Comment:
Are these meant to be commented out?

Reply:
well, they should be active once the second PR of UI modes is accepted, untill then, they are not acted on yet, so I left them here as placeholders, and to illustrate the extension I had in mind.

Reply:
Got it, thanks.